File: leetcode/3195. 包含所有 1 的最小矩形面积 I.go
func minimumArea(grid [][]int) int {
	r1 := len(grid[0])
	d1 := len(grid)
	r2 := 0
	c2 := 0
	for i := 0; i < len(grid); i++ {
		for j := 0; j < len(grid[i]); j++ {
			if grid[i][j] == 1 {
				r1 = Min(r1, j)
				d1 = Min(d1, i)
				r2 = Max(r2, j)
				c2 = Max(c2, i)
			}
		}
	}
	return (r2 - r1 + 1) * (c2 - d1 + 1)
}

func Min(a, b int) int {
	if a < b {
		return a
	}
	return b
}

func Max(a, b int) int {
	if a > b {
		return a
	}
	return b
}

File: leetcode/3195. 包含所有 1 的最小矩形面积 I.py
class Solution:
    def minimumArea(self, grid: List[List[int]]) -> int:
        r1 = len(grid[0])
        d1 = len(grid)
        r2 = -1
        d2 = -1
        for i in range(len(grid)):
            for j in range(len(grid[0])):
                if grid[i][j] == 1:
                    r1 = min(r1, j)
                    d1 = min(d1, i)
                    r2 = max(r2, j)
                    d2 = max(d2, i)
        return (r2 - r1 + 1) * (d2 - d1 + 1)